src/register-service-worker.js

```diff
@@ -0,0 +1,162 @@
+/* eslint-env browser */
+
+(function() {
+  var constants = require('./lib/constants.js');
+
+  var swScript = document.currentScript.dataset.serviceWorker;
+  var manifestAttribute = document.documentElement.getAttribute('manifest');
+
+  if (swScript && manifestAttribute && 'serviceWorker' in navigator) {
```

Comment:
Righto. I'll just do a conditional `register()` at the end of the Promise chain if `data-service-worker` is provided, and nothing if it's not provided.